crates/sui-core/src/authority_client.rs

```diff
@@ -102,6 +104,17 @@ impl NetworkAuthorityClient {
     }
 }
 
+#[async_trait]
+impl Reconfigurable for NetworkAuthorityClient {
+    fn needs_network_recreation() -> bool {
+        true
+    }
+
+    fn recreate(channel: tonic::transport::Channel) -> Self {
+        NetworkAuthorityClient::new(channel)
+    }
+}
```

Comment:
I don't think we need this. On reconfig we can just drop clients we don't want, hold onto ones that we still want, and create new clients for new validators.

Reply:
This is to be able to support LocalAuthorityClient (which doesn't use network) when constructing ActiveAuthority (which is used everywhere in tests)